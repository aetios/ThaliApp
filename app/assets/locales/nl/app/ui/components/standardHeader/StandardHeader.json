{
  "Welcome": "Welkom",
  "Event": "Evenement",
  "Calendar": "Agenda",
  "Pizza": "Pizza",
  "Profile": "Profiel",
  "Registration": "Registratie",
  "Settings": "Instellingen",
  "Notifications": "Meldingen"
}
