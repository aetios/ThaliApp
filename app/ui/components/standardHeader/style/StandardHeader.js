import { Platform, StatusBar } from 'react-native';

import StyleSheet from '../../../style/StyleSheet';
import Colors from '../../../style/Colors';

export const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : StatusBar.currentHeight;
export const APPBAR_HEIGHT = Platform.OS === 'ios' ? 44 : 56;

export const TOTAL_BAR_HEIGHT = APPBAR_HEIGHT + STATUSBAR_HEIGHT;

const styles = StyleSheet.create({
  statusBar: {
    height: STATUSBAR_HEIGHT,
    backgroundColor: Colors.magenta,
    android: {
      elevation: 4,
    },
  },
  appBar: {
    backgroundColor: Colors.magenta,
    height: APPBAR_HEIGHT,
    flexDirection: 'row',
    android: {
      height: APPBAR_HEIGHT,
      justifyContent: 'flex-start',
      alignItems: 'center',
      flexWrap: 'wrap',
      elevation: 4,
    },
    ios: {
      borderStyle: 'solid',
      borderBottomColor: Colors.darkMagenta,
      borderBottomWidth: 1,
      justifyContent: 'space-between',
      alignItems: 'center',
    },
  },
  title: {
    color: Colors.white,
    android: {
      fontSize: 20,
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontSize: 18,
      fontFamily: 'System',
      fontWeight: '600',
      textAlign: 'center',
    },
  },
  icon: {
    android: {
      paddingLeft: 20,
      paddingRight: 32,
    },
    ios: {
      paddingLeft: 10,
      paddingRight: 16,
    },
    color: Colors.white,
  },
  rightView: {
    ios: {
      width: 24 + 16 + 10,
      height: 0,
    },
  },
});

export default styles;
