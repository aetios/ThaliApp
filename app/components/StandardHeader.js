import React from 'react';
import { View, StatusBar, TouchableOpacity, Text } from 'react-native';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/MaterialIcons';

import { colors } from '../style';
import styles from './style/standardHeader';

import * as actions from '../actions/navigation';

const sceneToTitle = (scene, t) => {
  switch (scene) {
    case 'welcome':
      return t('Welcome');
    case 'event':
      return t('Event');
    case 'eventList':
      return t('Calendar');
    case 'pizza':
      return t('Pizza');
    case 'profile':
      return t('Profile');
    case 'registration':
      return t('Registration');
    default:
      return 'ThaliApp';
  }
};

const StandardHeader = props => (
  <View>
    <View style={styles.statusBar}>
      <StatusBar
        backgroundColor={colors.semiTransparent}
        translucent
        animated
        barStyle="light-content"
      />
    </View>
    <View style={styles.appBar}>
      <TouchableOpacity
        onPress={props.isFirstScene ? () => props.updateDrawer(!props.drawerOpen) : props.back}
      >
        <Icon
          name={props.isFirstScene ? 'menu' : 'arrow-back'}
          style={styles.icon}
          size={24}
        />
      </TouchableOpacity>
      <Text style={styles.title}>{sceneToTitle(props.currentScene, props.t)}</Text>
      <View style={styles.rightView} />
    </View>
  </View>
);

StandardHeader.propTypes = {
  isFirstScene: PropTypes.bool.isRequired,
  currentScene: PropTypes.string.isRequired,
  drawerOpen: PropTypes.bool.isRequired,
  back: PropTypes.func.isRequired,
  updateDrawer: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  isFirstScene: state.navigation.previousScenes.length === 0,
  currentScene: state.navigation.currentScene,
  drawerOpen: state.navigation.drawerOpen,
});

const mapDispatchToProps = dispatch => ({
  back: () => dispatch(actions.back()),
  updateDrawer: isOpen => dispatch(actions.updateDrawer(isOpen)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps)(translate('standardHeader')(StandardHeader));
