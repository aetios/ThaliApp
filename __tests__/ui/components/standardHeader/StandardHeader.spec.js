import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import StandardHeader from '../../../../app/ui/components/standardHeader/StandardHeader';
import reducer from '../../../../app/reducers';

jest.mock('react-navigation', () => ({
  withNavigation: component => component,
}));

const mockNavigation = {
  state: 'mock',
};

describe('StandardHeader component', () => {
  const mockStore = configureStore(reducer);
  const initialState = {
    navigation: {
      currentScene: 'home',
      previousScenes: [],
      loggedIn: false,
      drawerOpen: false,
    },
    session: {
      displayName: 'Lorem ipsum',
      photo: 'photo',
      token: 'token123',
    },
  };
  const store = mockStore(initialState);

  it('renders correctly', () => {
    const tree = renderer
      .create(<StandardHeader store={store} navigation={mockNavigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
