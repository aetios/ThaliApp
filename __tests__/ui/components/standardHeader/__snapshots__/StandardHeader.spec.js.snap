// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardHeader component renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#E62272",
        "height": 20,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E62272",
        "borderBottomColor": "#C2185B",
        "borderBottomWidth": 1,
        "borderStyle": "solid",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 24,
            },
            Object {
              "color": "#FFFFFF",
              "paddingLeft": 10,
              "paddingRight": 16,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î—’
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      ThaliApp
    </Text>
    <View
      style={
        Object {
          "height": 0,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;
